src/components/Tags.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import { MBTTAGS } from './TagSuggestions';
+
+//Dependencies for drag n drop tags with suggestions
+//npm install --save react-tag-input
+//npm install --save react-dnd@5.0.0
+//npm install --save react-dnd-html5-backend@3.0.2
+import { WithContext as ReactTags } from 'react-tag-input';
+import { stringify, v4 as uuid } from 'uuid';
+
+const suggestions = MBTTAGS.map(mbttags => {
+  return {
+    id: uuid(),
+    text: mbttags,
+  };
+});
+
+const KeyCodes = {
+  comma: 188,
+  enter: 13,
+};
+
+type tagstruct = {
+  rank: number;
+  segment: string;
+  tag: { createdAT: string; isDeleted: boolean; name: string; _id: string };
+  _id: string;
+};
+
+interface MyProps {
+  tags: { id: string; text: string }[];
+  seg: any;
+  rank: number;
+}
+
+interface MyState {
+  tags: any;
+  suggestions: any;
+  k: number;
+}
+
+const delimiters = [KeyCodes.comma, KeyCodes.enter];
+
+const capWords = (str: string) => {
+  if (!str) return 'Error';
+  let pieces = str.split(' ');
+  pieces.forEach((p, i) => (pieces[i] = p.charAt(0).toUpperCase() + p.substr(1)));
+  return pieces.join(' ');
+};
+
+class Tags extends React.Component<MyProps, MyState> {
```

Comment:
Note that this uses the class style of React components whereas most of the components in this app use the function style. Both are fine, however I have found that the function style components are much more flexible and lightweight and easier to manage.

Reply:
Sure, it can be done. We can test it out as it is before I write a new component.